training/controllers/all_controllers.py

```diff
@@ -8,8 +8,32 @@
 from training.models.users import User
 from training.app_init import bcrypt
 from markupsafe import escape
+from flask import g
+from functools import wraps
+from training.app_init import mail, Message
 app.secret_key = b'_5#y2L"F4Q8z\n\xec]/'
 
+@app.before_request
+def user_to_g():
+    username = sesion.get('username')
+    if username is not None:
+        # Suponemos que el usuario si tiene la session, siempre va a estar en la base de datos
+        user = session.query(User).filter_by(id=username).first()
+        g.user = user
+    return None
+
+
+def login_required(function):
+    @wraps(function)
+    #wrap.__name__ = function.__name__
+    def wrap(*args, **kwargs):
+        username = sesion.get('username')
+        if username is None:
+            return "Error, must be logged in"
```

Comment:
You should redirect to login or return a 401 status page

Reply:
It will be fixed in the next PR in the next branch.